### PR TITLE
sp-core: impl serde for some offchain types

### PR DESCRIPTION
**Pull request by yjhmelody**
I need it.

### CODE REVIEW

File: primitives/core/src/offchain/mod.rs [hunk 1]
```diff
@@ -195,6 +195,7 @@ pub struct OpaqueNetworkState {
 
 /// Simple blob to hold a `Multiaddr` without committing to its format.
 #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, PassByInner, TypeInfo)]
+#[cfg_attr(feature = "std", derive(serde::Serialize, serde::Deserialize))]
 pub struct OpaqueMultiaddr(pub Vec<u8>);
```
Comment (bkchr) [inline primitives/core/src/offchain/mod.rs:199]:
```suggestion
pub struct OpaqueMultiaddr(pub crate::Bytes);
```

We should do this when you want to derive serde.

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
@yjhmelody it doesn't compile.

File: primitives/core/src/offchain/mod.rs (hunk 1 above)
Comment (yjhmelody) [reply primitives/core/src/offchain/mod.rs]: Hi, `crate::Bytes` cause the problem

**Comment by yjhmelody** [comment]
@bkchr Some other problems ?

**Review by bkchr — APPROVED** [review]

**Pull request merged**